src/app/app-routing.module.ts

```diff
@@ -1,7 +1,31 @@
 import { NgModule } from "@angular/core";
+import { LayoutComponent } from "./layout/layout.component";
+import { HomeComponent } from "./home/home.component";
+import { ErrorComponent } from "./error/error.component";
 import { Routes, RouterModule } from "@angular/router";
 
-const routes: Routes = [];
+const routes: Routes = [
+  {
+    path: "",
+    component: LayoutComponent,
+    children: [
```

Comment:
I know this is how it's set up in Pramod's example code, but when I was looking at things, it seemed like we don't really need to have a layout component and define children for it, because the auto-generated app.component.html naturally fills the same role. It has some stuff that will show up on any page of the app, and then a router-outlet element that will show whatever component the user navigates too. I'm sure this will work too, but it might be a bit more complicated than necessary.